osu.Game/Screens/Play/RoomSubmittingPlayer.cs

```diff
@@ -14,15 +13,18 @@ namespace osu.Game.Screens.Play
     /// </summary>
     public abstract class RoomSubmittingPlayer : SubmittingPlayer
     {
-        [Resolved(typeof(Room), nameof(Room.RoomID))]
-        protected Bindable<long?> RoomId { get; private set; }
+        protected readonly IBindable<long?> RoomId = new Bindable<long?>();
 
         protected readonly PlaylistItem PlaylistItem;
+        protected readonly Room Room;
 
-        protected RoomSubmittingPlayer(PlaylistItem playlistItem, PlayerConfiguration configuration = null)
+        protected RoomSubmittingPlayer(Room room, PlaylistItem playlistItem, PlayerConfiguration configuration = null)
             : base(configuration)
         {
+            Room = room;
             PlaylistItem = playlistItem;
+
+            RoomId.BindTo(room.RoomID);
```

Comment:
Rather than keeping the bindable flow here, would it make more sense to just consume immediately without binding? I think it's guaranteed this will be in a good state at this point (and the previous flow was just there because that was the "correct" way of accessing this id).

Reply:
That's fair. How do you feel about `Room.RoomID.Value`?

Reply:
Yeah, just at the point of usage. Passing in the room is still fine, just remove the binding portion.